apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -487,7 +488,15 @@ func (api *ProvisionerAPI) machineLXDProfileNames(
 			return nil, errors.Trace(err)
 		}
 
+		// TODO(nvinuesa): this call will fail with [applicationerrors.CharmNotFound]
+		// when trying to retrieve the LXD profiles. It happens because the
+		// charm hasn't been inserted into dqlite yet at this point in time
+		// (it hasn't started downloading).
```

Comment:
Is this because it's not activated?

Reply:
I might be wrong here, but the thing is that in order to get the lxd profile we need to download the charm, and this doesn't happen before provisioning. Right?

Reply:
Done the right fix, which is to pass the reference name into the charm locator struct.